core/src/main/java/org/apache/iceberg/deletes/EmptyPositionDeleteIndex.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.deletes;
+
+class EmptyPositionDeleteIndex implements PositionDeleteIndex {
+
+  private static final EmptyPositionDeleteIndex INSTANCE = new EmptyPositionDeleteIndex();
+
+  private EmptyPositionDeleteIndex() {}
+
+  static EmptyPositionDeleteIndex get() {
+    return INSTANCE;
+  }
+
+  @Override
+  public void delete(long position) {
+    throw new UnsupportedOperationException("Cannot modify " + getClass().getName());
+  }
+
+  @Override
+  public void delete(long posStart, long posEnd) {
+    throw new UnsupportedOperationException("Cannot modify " + getClass().getName());
+  }
+
+  @Override
+  public boolean isDeleted(long position) {
+    return false;
+  }
+
+  @Override
+  public boolean isEmpty() {
+    return true;
+  }
+
+  @Override
+  public boolean equals(Object other) {
+    if (this == other) {
```

Comment:
Since this is a static class isn't this enough? return this == other?

Reply:
I guess it may be serialized and deserialized to get another instance of the class.